code/Data_Analysis.ipynb

```diff
@@ -1096,143 +1096,285 @@
+    "    space = 1\n",
+    "    label = format(y_value)\n",
+    "    ax.annotate(label, (x_value, y_value), xytext=(0, space), textcoords=\"offset points\", ha='center', va='bottom')    \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77a34dea",
+   "metadata": {},
+   "source": [
+    "The plot shows that there is no even distribution and the diagnosis with the number 2 does not even occur in the data set used here. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94c3a728",
+   "metadata": {},
+   "source": [
+    "For the further steps we need the group variable in a np.array format. So, it is now formatted accordingly."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 52,
+   "execution_count": 75,
    "id": "dadcac26",
    "metadata": {},
    "outputs": [],
    "source": [
     "groups = np.asarray(groups)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "46a0bf6f",
+   "metadata": {},
+   "source": [
+    "Now, cross-validation is used to compare the connectomes across the groups and check the predictive power. For this purpose, the participants will be assigned to training and test sets using 'KFold'. This allows splitting the dataset into k consecutive folds. Each fold is then used once as a validation while the k-1 remaining folds form the training set. Let's do this for 10 folds and have a look at the splits."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": 48,
-   "id": "91ebbef1",
+   "execution_count": 88,
+   "id": "395ad3d8",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TRAIN: [0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [1 1 0 0]\n",
+      "TRAIN: [1 1 0 0 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [1 1 3 3]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 0 3 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 3 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 0 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 1 1 0] TEST: [0 3 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0] TEST: [0 1 1 0]\n"
+     ]
+    }
+   ],
    "source": [
-    "cv = StratifiedShuffleSplit(n_splits=30, random_state=0, test_size=10)"
+    "from sklearn.model_selection import KFold\n",
+    "kf = KFold(n_splits=10)\n",
```

Comment:
I think the prior CV was working fine and gave you the option to create more splits based on shuffling, which could be useful going forward.